### ISSUE TITLE
No templates matched the input template name: blazorwasm.

### ISSUE THREAD

**Issue by FANMixco** [aspnet/AspNetCore.Docs#14547]
# Issue description

I get this issue when I'm trying to create a new project in VS Code for Mac and I cannot get the templates despite, I already have the latest version.

> No templates matched the input template name: blazorwasm.

# Software versions

Check the .NET target framework(s) being used, and include the version number(s).

- [x] .NET Core
- [ ] .NET Framework
- [ ] .NET Standard

If using the .NET Core SDK, include `dotnet --info` output. If using .NET Framework without the .NET Core SDK, include info from Visual Studio's **Help** > **About Microsoft Visual Studio** dialog.

> .NET Core SDK
> SDK: /usr/local/share/dotnet/sdk/3.0.100/Sdks
> SDK Versions:
> 	3.0.100
> 	2.1.701

**Comment by scottaddie** [comment]
@FANMixco Have you installed the project templates that shipped yesterday? Run the following command:

```
dotnet new -i Microsoft.AspNetCore.Blazor.Templates::3.0.0-preview9.19465.2
```

**Comment by FANMixco** [comment]
Hi @scottaddie , I tried it and it didn't work.

<img width="1280" alt="Screen Shot 2019-09-24 at 23 19 06" src="https://user-images.githubusercontent.com/1666156/65551301-e6c38980-df21-11e9-8710-0818dfe02d71.png">
<img width="1280" alt="Screen Shot 2019-09-24 at 23 19 13" src="https://user-images.githubusercontent.com/1666156/65551302-e6c38980-df21-11e9-8585-55307287156c.png">
<img width="1280" alt="Screen Shot 2019-09-24 at 23 19 34" src="https://user-images.githubusercontent.com/1666156/65551303-e75c2000-df21-11e9-928a-9933cd3c0360.png">
<img width="1280" alt="Screen Shot 2019-09-24 at 23 19 37" src="https://user-images.githubusercontent.com/1666156/65551304-e75c2000-df21-11e9-8f3c-1fa341984deb.png">
<img width="1280" alt="Screen Shot 2019-09-24 at 23 19 49" src="https://user-images.githubusercontent.com/1666156/65551305-e7f4b680-df21-11e9-935d-26683d88b080.png">
<img width="1280" alt="Screen Shot 2019-09-24 at 23 19 55" src="https://user-images.githubusercontent.com/1666156/65551306-e7f4b680-df21-11e9-9f00-b0a92c1b9264.png">

Also, this is my configuration of VS for Mac:

> Visual Studio Community 2019 for Mac
> Version 8.3 (build 1805)
> Installation UUID: 85bb6e8f-2711-4ac6-a750-7597235180eb
> 	GTK+ 2.24.23 (Raleigh theme)
> 	Xamarin.Mac 5.16.1.24 (d16-3 / 08809f5b)
> 
> 	Package version: 604000198
> 
> Mono Framework MDK
> Runtime:
> 	Mono 6.4.0.198 (2019-06/fe64a4765e6) (64-bit)
> 	Package version: 604000198
> 
> NuGet
> Version: 5.3.0.6192
> 
> .NET Core SDK
> SDK: /usr/local/share/dotnet/sdk/3.0.100/Sdks
> SDK Versions:
> 	3.0.100
> 	2.1.701
> MSBuild SDKs: /Library/Frameworks/Mono.framework/Versions/6.4.0/lib/mono/msbuild/Current/bin/Sdks
> 
> .NET Core Runtime
> Runtime: /usr/local/share/dotnet/dotnet
> Runtime Versions:
> 	3.0.0
> 	2.1.13
> 
> Xamarin.Profiler
> Version: 1.6.11.16
> Location: /Applications/Xamarin Profiler.app/Contents/MacOS/Xamarin Profiler
> 
> Updater
> Version: 11
> 
> Xamarin.Android
> Version: 10.0.0.43 (Visual Studio Community)
> Commit: xamarin-android/d16-3/8af1ca8
> Android SDK: /Users/fanmixco/Library/Developer/Xamarin/android-sdk-macosx
> 	Supported Android versions:
> 		8.1 (API level 27)
> 
> SDK Tools Version: 26.1.1
> SDK Platform Tools Version: 29.0.2
> SDK Build Tools Version: 28.0.3
> 
> Build Information: 
> Mono: mono/mono/2019-06@7af64d1ebe9
> Java.Interop: xamarin/java.interop/d16-3@5836f58
> LibZipSharp: grendello/LibZipSharp/d16-3@71f4a94
> LibZip: nih-at/libzip/rel-1-5-1@b95cf3f
> ProGuard: xamarin/proguard/master@905836d
> SQLite: xamarin/sqlite/3.27.1@8212a2d
> Xamarin.Android Tools: xamarin/xamarin-android-tools/d16-3@cb41333
> 
> Microsoft Mobile OpenJDK
> Java SDK: /Users/fanmixco/Library/Developer/Xamarin/jdk/microsoft_dist_openjdk_1.8.0.25
> 1.8.0-25
> Android Designer EPL code available here:
> https://github.com/xamarin/AndroidDesigner.EPL
> 
> Android SDK Manager
> Version: 1.4.0.65
> Hash: c33b107
> Branch: remotes/origin/d16-3
> Build date: 2019-09-19 20:42:44 UTC
> 
> Android Device Manager
> Version: 1.2.0.115
> Hash: 724ea69
> Branch: remotes/origin/d16-3
> Build date: 2019-09-19 20:43:06 UTC
> 
> Xamarin Inspector
> Version: 1.4.3
> Hash: db27525
> Branch: 1.4-release
> Build date: Mon, 09 Jul 2018 21:20:18 GMT
> Client compatibility: 1
> 
> Xamarin Designer
> Version: 16.3.0.230
> Hash: bbe518670
> Branch: remotes/origin/d16-3-xcode11
> Build date: 2019-09-17 18:39:30 UTC
> 
> Apple Developer Tools
> Xcode 11.0 (14936)
> Build 11A420a
> 
> Xamarin.Mac
> Version: 6.2.0.42 (Visual Studio Community)
> Hash: 5e8a208b
> Branch: d16-3
> Build date: 2019-09-17 14:19:30-0400
> 
> Xamarin.iOS
> Version: 13.2.0.42 (Visual Studio Community)
> Hash: 5e8a208b
> Branch: d16-3
> Build date: 2019-09-17 14:19:30-0400
> 
> Build Information
> Release ID: 803001805
> Git revision: df50511ccb08d4d84a53939f1d7fc0b40cbb0ee1
> Build date: 2019-09-20 16:10:55+00
> Build branch: release-8.3
> Xamarin extensions: da8d448cd65f2cc87c5e6dc7e2c2ca7b96d20c72
> 
> Operating System
> Mac OS X 10.14.6
> Darwin 18.7.0 Darwin Kernel Version 18.7.0
>     Tue Aug 20 16:57:14 PDT 2019
>     root:xnu-4903.271.2~2/RELEASE_X86_64 x86_64
>

**Comment by guardrex** [comment]
@FANMixco You said "VS Code for Mac" ... but I think you meant VS4Mac in your original post.

They aren't available within VS4Mac at this time. We don't have an ETA. They are working on it. One way that you can keep tabs on when it is available is to keep an eye on our *Get Started* topic at https://docs.microsoft.com/aspnet/core/blazor/get-started ... at Item 3 ... the tab control for the various tooling. When you see me add a **Visual Studio for Mac** tab there, then you know it's available.

WRT VSC ... if you decide to use that tooling ... create the project in a command shell and then you should be able to open and work with it in VSC. I can't attest to the debugger running it ok. You might need to (temporarily) run such apps from a command shell (`dotnet run`).

**Comment by FANMixco** [comment]
> @FANMixco You said "VS Code for Mac" ... but I think you meant VS4Mac in your original post.
> 
> They aren't available within VS4Mac at this time. We don't have an ETA. They are working on it. One way that you can keep tabs on when it is available is to keep an eye on our _Get Started_ topic at https://docs.microsoft.com/aspnet/core/blazor/get-started ... at Item 3 ... the tab control for the various tooling. When you see me add a **Visual Studio for Mac** tab there, then you know it's available.
> 
> WRT VSC ... if you decide to use that tooling ... create the project in a command shell and then you should be able to open and work with it in VSC. I can't attest to the debugger running it ok. You might need to (temporarily) run such apps from a command shell (`dotnet run`).

Well, here is the bug from VS Code:

<img width="1280" alt="Screen Shot 2019-09-25 at 19 14 03" src="https://user-images.githubusercontent.com/1666156/65623731-bdaa0400-dfc8-11e9-9abe-754469086930.png">

**Comment by guardrex** [comment]
Let me see if I can repro that here. I'll fire up my Mac and see if I get the same thing.

Give me ... er ... 15 minutes. I'll get back to you.

**Comment by guardrex** [comment]
`blazorwasm` isn't in that list ... but in your earlier screenshot, I see it ...

![image](https://user-images.githubusercontent.com/1622880/65624079-be25a900-df8e-11e9-815f-71346f511b95.png)

You may need to run the template install command again.

**Comment by FANMixco** [comment]
> `blazorwasm` isn't in that list ... but in your earlier screenshot, I see it ...
> 
> ![image](https://user-images.githubusercontent.com/1622880/65624079-be25a900-df8e-11e9-815f-71346f511b95.png)
> 
> You may need to run the template install command again.

From what I see I need to run it as an Admin, please add that update in the documentation since I can only run it if I am admin.

**Comment by guardrex** [comment]
Clarify that ... run *what?* ... the template install command or the command that creates an app from the template?

**Comment by guardrex** [comment]
U sure? ... I was just able to install the templates as the guest user on my Mac.

**Comment by guardrex** [comment]
... and create a new Blazor WebAssembly app from that template ... as the guest user. I can't repro your scenario here.

**Comment by FANMixco** [comment]
> ... and create a new Blazor WebAssembly app from that template ... as the guest user. I can't repro your scenario here.

Yep, do you want that I record my screen? I'm the admin.

**Comment by guardrex** [comment]
Open an engineering issue. I'm not sure what the problem might be. I can't repro it here.

https://github.com/aspnet/AspNetCore/issues

If you don't mind, add a "cc: @guardrex" at the bottom so that it will ping me and I can follow the discussion.

**Comment by soccerjoshj07** [comment]
> @FANMixco Have you installed the project templates that shipped yesterday? Run the following command:
> 
> ```
> dotnet new -i Microsoft.AspNetCore.Blazor.Templates::3.0.0-preview9.19465.2
> ```

Thanks @scottaddie , this worked for me :)

**Comment by guardrex** [comment]
@FANMixco That version is no longer supported and the templates are rolled into the 3.1.300 SDK that just released. Use the released bits with the 3.1.300 SDK, which is supported ...

https://dotnet.microsoft.com/download/dotnet-core/3.1
